src/js/components/message.jsx

```diff
@@ -43,43 +43,49 @@ export class MessageComponent extends Component {
     }
 
     render() {
-        const {name, role, mediaUrl, avatarUrl, text, accentColor, firstInGroup, lastInGroup, linkColor} = this.props;
+        const {name, role, avatarUrl, text, accentColor, firstInGroup, lastInGroup, linkColor, type} = this.props;
         const actions = this.props.actions.filter((a) => a.type !== 'reply');
 
+        const hasText = text && text.trim();
+        const hasImage = type === 'image';
+        const isAppUser = role === 'appUser';
+        const hasActions = actions.length > 0;
+
+        const avatarClass = hasImage ? ['sk-msg-avatar', 'sk-msg-avatar-img'] : ['sk-msg-avatar'];
+        const avatarPlaceHolder = isAppUser ? null : (<div className='sk-msg-avatar-placeholder' />);
+        const containerClass = [];
+
+        if (hasImage || actions.length > 0) {
+            containerClass.push('sk-msg-image');
+        }
+
+        if (hasText || hasActions) {
+            containerClass.push('sk-msg');
+        }
+
         const actionList = actions.map((action) => {
             return <Action key={ action._id }
                            buttonColor={ linkColor }
                            {...action} />;
         });
 
-        const isAppUser = role === 'appUser';
-
-        const avatarClass = mediaUrl ? ['sk-msg-avatar', 'sk-msg-avatar-img'] : ['sk-msg-avatar'];
-
-        const avatar = isAppUser ?
-            null :
+        const avatar = isAppUser ? null :
             <img className={ avatarClass.join(' ') }
                  src={ avatarUrl } />;
 
-        const avatarPlaceHolder = isAppUser ? null : (<div className='sk-msg-avatar-placeholder' />);
-
-        const message = mediaUrl ?
-            <ImageMessage {...this.props}
-                          actions={ actions } /> :
-            <TextMessage {...this.props}
-                         actions={ actions } />;
-
-        const containerClass = [mediaUrl ? 'sk-msg-image' : 'sk-msg'];
-
-        const hasContent = (text && text.trim()) || (mediaUrl && mediaUrl.trim());
+        const textPart = hasText && <TextMessage {...this.props}
+                                                 hasActions={ hasActions }
+                                                 hasImage={ hasImage } />;
+        const imagePart = hasImage && <ImageMessage {...this.props}
```

Comment:
same thing here, I think we should just pass the necessary styling to the component instead of making them aware of the concept of actions. (I know it was like that before, but I think it was wrong).